### PR TITLE
test (e2e): Add new e2e testRigor tests

### PR DESCRIPTION
**Pull request by amandassa**
## Description
Added e2e tests for the main web-app functionalities.

### Dependencies
- [testRigor CLI](https://testrigor.com/command-line)

## Related issues and discussion
#2199

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] If you have multiple commits please combine them into one commit by squashing them.

- [X] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by IOhacker** [comment]
@amandassa are you able to fix the conflict file?

**Comment by amandassa** [comment]
Hello @IOhacker, can you review the PR please?

**Comment by IOhacker** [comment]
@amandassa I think that this PR requires a suscription for a paid service? isn't ?

**Comment by amandassa** [comment]
@IOhacker For this integration I used a testRigor OpenSource suite. The testRigor OpenSource option provides a free suite and it is free forever. You can get more info at [testRigor Website](https://testrigor.com/)

**Comment by IOhacker** [comment]
Which is the open source github repository?

Where can we review the license?

I see some variables that must be enable. Which are the steps to get the
values and set them on the github account.

How to set the values for a developer? Or how to skip this testing?

Is there any other similar alternatives?

Do you have a video about the changes/benefits of this PR?

Regards

Regards

Victor

**Comment by amandassa** [comment]
@IOhacker 

Q: Which is the open source github repository? Where can we review the license? 

**A: testRigor OpenSource is a free testRigor account option that allows testing open source projects using testRigor service for free.**

Q: I see some variables that must be enable. Which are the steps to get the values and set them on the github account. How to set the values for a developer? 

**A: The variables CI/CD Token and test suite ID can be found at the test suite CI integration page and test suite initial page.** 

**For this PR, I created a free testRigor account and suite for Mifos and linked the free suite to the bash script that runs the testRigor CLI tool on the workflow. This means that no further configuration is needed, since the workflow is already triggering a testRigor run within the free suite I created.**

**In addition to the integration, I included a README.md file with instructions for integration for Mifos developers. Following those instructions it is possible to test new features locally with testRigor before they are deployed. It is also possible for a developer to create their own testRigor account and test locally on that. It is also possible to change the variables and link the execution to a test suite different than mine.** 

Q: Or how to skip this testing? 

**A: This testing step consists of reliable End-to-end testing of the most crucial functionalities in Mifos. Currently only the Mifos login page is being tested. Including these tests to the CI would drastically improve reliability and bug tracking**

Q: Is there any other similar alternatives? Do you have a video about the changes/benefits of this PR?

**A: One of the reasons for using testRigor instead of other testing alternatives like Cypress is the improved maintenability. Since the tests are written in plain english, anyone can create and update tests. That can be useful for an open source project like Mifos. I noticed you have a directory for cypress integration tests, but no tests were added in the last 3 years. 
You can read further in [testRigor Benefits Video Demonstration](https://www.youtube.com/watch?v=GYLVwXLu564) or [testRigor Benefists Page](https://testrigor.com/benefits/)**

**Comment by IOhacker** [comment]
Hello @amandassa,

I have look for the license and terms of use and I have only found these:

https://testrigor.com/terms-of-use/
https://choosealicense.com/licenses/unlicense/

I was unable to find any github repository for libraries or source code of RigorTest. 

In my opinion I am not fully convinced to link a commercial software testing platform (even with a Free account). I have not found how this platform is supporting the Open Source communities.

I reviewed the scripts included in the PR and they implement the functionality provided by RigorTest Platform which could lead to a vendor lock-in.

**Comment by DavidH-1** [comment]
Thanks @IOhacker for highlighting. In some ways I’m not worried by plain text definition that has advantages just like plan text in Cucumber/gherkin I guess my concerns are two fold:
(A) from a quick read and it is very quick it appears that testrigor then takes that plan text parses it through an LLM to generate the test. How repeatable is the test then? As the LLM is trained over time will the test outcome for same input change. I need to research it more for this.
(B) does merging this mandate the use of testrigor, if so then all the licensing queries etc become relevant and this should be referred to the architecture working group for consideration. If it’s just a set of optional additional tests that could be run then maybe it’s of less concern.

**Comment by amandassa** [comment]
@DavidH-1Thanks for reviewing.
Regarding your questions:
(A) The specific test steps are parsed and executed emulating how a human would do it, without AI. AI is used to build steps on demand, if an unrecognized phrase is provided.
(B) The integration implies only on the installation of the CLI tool. The tests are then run on testRigor through tunneling using the same deployment as the other workflow steps.
